src/Settings/RadioSetting.svelte

```diff
@@ -0,0 +1,56 @@
+<script lang="ts">
+    import RadioButton from './RadioButton.svelte'
+    import ExpandMoreIcon from './expand-more-icon.svg'
+
+    import { createEventDispatcher } from 'svelte'
+
+    const dispatch = createEventDispatcher()
+    type OptionType = {
+        value: string
+        text: string
+    }
+    export let label: string
+    export let options: OptionType[]
+    export let currentSettingValue: string
+    export let optionName: string
+
+    let newSettingValue = currentSettingValue
+
+    $: dispatch('settingChange', newSettingValue)
+    $: currentSettingValueLabel = options?.find(option => option.value === currentSettingValue)?.text ?? ''
+    let isModeExpanded = false
+</script>
+
+<style>
+    .height-transition {
+        transition: max-height 0.5s;
+    }
+
+    .expanded {
+        max-height: 80px;
```

Comment:
Can we have something different from hardcoded 80px? Theoratically this makes component not easily reusable